lib/gcloud/datastore/key.rb

```diff
@@ -226,41 +224,43 @@ def incomplete?
       end
 
       ##
-      # @private Convert the Key to a protocol buffer object.
-      def to_proto
-        Proto::Key.new.tap do |k|
-          k.path_element = path.map do |pe_kind, pe_id_or_name|
-            Proto.new_path_element pe_kind, pe_id_or_name
+      # @private Convert the Key to a Google::Datastore::V1beta3::Key object.
+      def to_grpc
+        grpc_path = path.map do |pe_kind, pe_id_or_name|
+          path_args = { kind: pe_kind }
+          if pe_id_or_name.is_a? Integer
+            path_args[:id] = pe_id_or_name
+          else
+            path_args[:name] = pe_id_or_name
           end
-          k.partition_id = Proto.new_partition_id dataset_id, namespace
+          Google::Datastore::V1beta3::Key::PathElement.new(path_args)
+        end
+        grpc = Google::Datastore::V1beta3::Key.new(path: grpc_path)
+        if dataset_id || namespace
```

Comment:
@pcostell Yes, absolutely. So far this PR has no breaking ruby API changes, and is only swapping out the underlying communication to the service. We have a few small changes to the ruby API we'd like to make, and renaming `dataset_id` on Key is one of them. Do you want us to include them in this PR or should we make them in a separate PR?